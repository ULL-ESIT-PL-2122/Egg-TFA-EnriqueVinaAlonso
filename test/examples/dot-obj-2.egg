do (
  def(x, object ( 
    c:   0,
    gc:  ->(self.c),
    sc:  ->(value, =(self["c"], value)),
    inc: ->(=(self["c"], +(self.c, 1)))
  )),
  print(x),
  print(x["c"]),           
  print(x.c),             
  print(x.gc()),           
  print(element(x, "gc")), 
  print(x["gc"]),   
  print(x["gc"]),     
  x.sc(5),                 
  print(x.gc()),           
  x["sc"](9),
  print(x.gc())            
)

