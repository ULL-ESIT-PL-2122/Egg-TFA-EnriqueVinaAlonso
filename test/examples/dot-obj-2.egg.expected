{"c":0}
0
0
0
function(...args) {
function () { [native code] }
function () { [native code] }
5
9